echo/wavefront-tracing-utils.go

```diff
@@ -0,0 +1,173 @@
+package echo
+
+import (
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/labstack/echo"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	reporter "github.com/wavefronthq/wavefront-opentracing-sdk-go/reporter"
+	"github.com/wavefronthq/wavefront-opentracing-sdk-go/tracer"
+	wfTracer "github.com/wavefronthq/wavefront-opentracing-sdk-go/tracer"
+	wfApplication "github.com/wavefronthq/wavefront-sdk-go/application"
+	"github.com/wavefronthq/wavefront-sdk-go/senders"
+)
+
+// Tracer Global Tracer
+var Tracer opentracing.Tracer
+var globalTracerConfig *tracerConfig
+
+// InitTracer initialize tracer object, initialize the GlobalTracer with this newly created,
+// tracer object.
+// Call utils/wavefront.Tracer to access the tracer object
+//For proxy sender send token as empty string
+func InitTracer(configFilePath string, routesRegistrationFilePath string, echoWeb *echo.Echo, sendDirectly bool, server string, token string, flushIntervalSeconds int) error {
+
+	//Constructing globalTracerConfig from config file
+	tracerConfig, err := readConfigFromYaml(configFilePath)
+	if err != nil {
+		return err
+	}
+	//Registering routes
+	tracerConfig, err = readConfigFromYaml(routesRegistrationFilePath)
+	if err != nil {
+		return err
+	}
+
+	var sender senders.Sender
+	if sendDirectly {
+		// Create the direct sender
+		sender, err = getWavefrontDirectSender(server, token, flushIntervalSeconds)
+		if err != nil {
+			return err
+		}
+	} else {
+		// Create the proxy sender
+		sender, err = getWavefrontProxySender(server, flushIntervalSeconds)
+		if err != nil {
+			return err
+		}
+	}
+	appTags := wfApplication.New(tracerConfig.Application, tracerConfig.Service)
+	if tracerConfig != nil {
+		if len(tracerConfig.Cluster) > 0 {
+			appTags.Cluster = tracerConfig.Cluster
+		}
+		if len(tracerConfig.Shard) > 0 {
+			appTags.Shard = tracerConfig.Shard
+		}
+		if tracerConfig.CustomApplicationTags.StaticTags != nil && len(tracerConfig.CustomApplicationTags.StaticTags) > 0 {
+			for k, v := range tracerConfig.CustomApplicationTags.StaticTags {
+				appTags.CustomTags[k] = v
+			}
+		}
+	}
+	wfReporter := reporter.New(sender, appTags, reporter.Source(tracerConfig.Source))
+	clReporter := reporter.NewConsoleSpanReporter(tracerConfig.Source)
```

Comment:
the console and composite reporters are typically only used for debugging purposes. We can remove these and go with just the `wfReporter` instance.